### PR TITLE
Nopatch lua CVE-2020-15889

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
`lua` CVE-2020-15889 does not affect Lua 5.3.5. The vulnerability is introduced in Lua 5.4.0.

###### Change Log  
- Nopatch `lua` CVE-2020-15889 

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2020-15889 

###### Test Methodology

- make 'input-srpms' to validate spec file format.

### CODE REVIEW

**Comment by lpereira** [comment]
From my reading of https://nvd.nist.gov/vuln/detail/CVE-2020-15889, this affects all versions of Lua up to (and including) 5.4.0:

> Lua through 5.4.0 has a getobjname heap-based buffer over-read because youngcollection in lgc.c uses markold for an insufficient number of list members.

So it might be prudent to use the upstream patch: https://github.com/lua/lua/commit/127e7a6c8942b362aa3c6627f44d660a4fb75312

**Comment by lpereira** [comment]
I looked at Lua 5.3.5, and the offending code doesn't seem to be there. Looking closer.

**Comment by dmcilvaney** [comment]
https://github.com/lua/lua/commit/127e7a6c8942b362aa3c6627f44d660a4fb75312 Modifies code that doesn't even exist in our version. As best I can tell this is an oversight with NVD. There is a new feature (generational collector) started here: https://github.com/lua/lua/commit/f5f3df3bd17fb3489bbd26ab39fe1580a8dbf9c9 which applies to the 5.4.0 branch only.

**Review by lpereira — APPROVED** [review]
:shipit:

File: SPECS/lua/CVE-2020-15889.nopatch [hunk 1]
```diff
@@ -0,0 +1 @@
+# CVE-2020-15889 is in the Lua generational garbage collection code, which is new to 5.4.0. 3.3.5 is not affected.
```
Comment (MateuszMalisz) [inline SPECS/lua/CVE-2020-15889.nopatch:1]: typo: s:`3.3.5`:`5.3.5`

File: SPECS/lua/lua.spec [hunk 2]
```diff
@@ -13,6 +13,8 @@ Distribution:   Mariner
 Source0:        https://www.lua.org/ftp/%{name}-%{version}.tar.gz
 Source1:        %{LICENSE_PATH}
 Patch0:         lua-5.3.4-shared_library-1.patch
+# CVE-2020-15889 is in the Lua generational garbage collection code, which is new to 5.4.0. 3.3.5 is not affected.
```
Comment (MateuszMalisz) [inline SPECS/lua/lua.spec:16]: typo as well

**Review by MateuszMalisz — APPROVED** [review]
Please address the typos

**Pull request merged**